### ISSUE TITLE
add pre-processed license.h

### ISSUE THREAD

**Issue by glensc** [krallin/tini#62]
I do not usually suggest this, but perhaps it's reasonable to include the license header file in git repo?

the `xxd` requirement is rather uncommon, and installing it is troublesome considering how differently distributions package that vim package addon (`xxd` comes from vim sources).

this avoids issues like: https://github.com/docker/docker/issues/28548

other option would be to upload release tarball to issue, like you upload built binaries. then they can just use the tar.gz which includes the generated header file.

**Comment by krallin** [comment]
Thanks for the heads up @glensc,

@crosbymichael, @mlaventure: do you have a bit of insight into whether there are other folks building Docker that are likely to run issues due to this? I'm happy to try and change things up to remove the dependency on `xxd`, but before I do so I'd like to know whether this is likely to continue causing trouble moving forward (as opposed to said trouble being behind us).

Thanks!

**Comment by crosbymichael** [comment]
I would be nice to have it included in a headerfile and remove the `vim-common` dep because its super weird that we have to install vim-common to compile a small init

**Comment by glensc** [comment]
@krallin my second suggestion, to publish release tarballs should work too, then you don't have to add generated content to git repo.

**Comment by krallin** [comment]
@glensc I *think* the Docker folks prefer building off of a commit (if only from an auditing perspective); I'll let @crosbymichael confirm though.

**Comment by glensc** [comment]
@kralin if you make releases often they don't need commit but can use tags.

**Comment by crosbymichael** [comment]
Building off of a commit or tag is good for us.  As if there is a bug fix that we need or something like that we don't have to wait on @krallin to create new binaries ;)  We don't want to be a burden so building from source or if we want to compile it with apline to create smaller binaries we can.

**Comment by glensc** [comment]
@crosbymichael binaries are created by travis, so @krallin needs to only make tag. but indeed, travis can be slow, it may take even 15 minutes to get the binary since the tagging.

**Comment by krallin** [comment]
@glensc from an auditing / security perspective, I think it's preferable for Docker to build their own binaries: I appreciate the added security of Docker depending on commit hashes (which provide some form of content-addressability).

**Comment by glensc** [comment]
`.tar.gz` + `.tar.gz.asc` can be verified too :)

anyway, i'm not here to convince you one or another way.

**Comment by krallin** [comment]
@glensc yes; I'm aware of that — however, I think it's best that Docker's security does not depend on my ability to safeguard my gpg key 😉

(I'm not saying saying I'm planning to leak my gpg key to everyone, but saying that we should avoid creating a vulnerability where none needs to exist)

**Comment by glensc** [comment]
aside, you should at least sign your tags:

https://help.github.com/articles/signing-commits-using-gpg/

**Comment by krallin** [comment]
Sure; thanks for your feedback.

**Comment by krallin** [comment]
Fixing this in https://github.com/krallin/tini/pull/63; thanks for letting me know this was an issue @glensc @crosbymichael 😄
